### PR TITLE
[feature/otx] Changed activation naming for proper checkpoint handling, fixed unexpected accuracy range

### PR DESCRIPTION
**Pull request by kprokofi**
Problem:
1) The checkpoint in OTX eval loaded incorrectly due to the `load_state_dict_pre_hook` function that changes keys in state_dict.
2) The accuracy range is expected to be in [0,1] range. mmcls computes in different range. This is actually just a copy-paste from develop branch

### CODE REVIEW

**Comment by JihwanEom** [comment]
Could you check pre-commit tests?

File: otx/mpa/modules/models/heads/custom_multi_label_non_linear_cls_head.py [hunk 1]
```diff
@@ -41,18 +41,17 @@ def __init__(
...
         if self.num_classes <= 0:
             raise ValueError(f"num_classes={num_classes} must be a positive integer")
 
-        self._init_layers()
+        self._init_layers(act_cfg)
 
-    def _init_layers(self):
-        modules = [nn.Linear(self.in_channels, self.hid_channels), nn.BatchNorm1d(self.hid_channels), self.act]
+    def _init_layers(self, act_cfg):
+        modules = [nn.Linear(self.in_channels, self.hid_channels), nn.BatchNorm1d(self.hid_channels), build_activation_layer(act_cfg)]
```
Comment (sungmanc) [inline otx/mpa/modules/models/heads/custom_multi_label_non_linear_cls_head.py]: Is this remedy just targetted to `multi_label_non_linear_cls_head.py`? what about `linear_cls_head`?

Comment (kprokofi) [reply otx/mpa/modules/models/heads/custom_multi_label_non_linear_cls_head.py]: In the multi-class classification we don't have such problem, since don't use PRelu activation. 
In general we need to get rid of awful implementation of state dict hooks. There are a lot of hardcoded things and it can shock  someone who reading this

**Comment by kprokofi** [comment]
I don't understand why black test doesn't pass. 
I reformatted both files with black functionality and test still failed...

**Comment by kprokofi** [comment]
Pre-merge checks passed
Codacy failed due to inconsistency between black and Codacy Static Code Analysis. Codacy requires 4 additional spaces, 
image
But, black requires as is now. 
I think, we could disregard this minor thing

**Comment by sungmanc** [comment]
@JihwanEom , could you check this again for code level? In my understanding, you are in charge of Classification task (especially, metric and performance). If not, please ignore :)

**Comment by sungmanc** [comment]
@kprokofi , CI tests FAILED due to the time limit, could you check with the local test?

**Comment by JihwanEom** [comment]
Thanks @sungmanc, I left some comment.

File: otx/mpa/modules/datasets/cls_dir_dataset.py [hunk 2]
```diff
@@ -186,6 +198,9 @@ def evaluate(self, results, metric="accuracy", metric_options=None, logger=None)
             self.class_acc = True
 
         eval_results = super().evaluate(results, metrics, metric_options, logger)
+        for k in metric_options["topk"]:
```
Comment (JihwanEom) [inline otx/mpa/modules/datasets/cls_dir_dataset.py]: Do we need to update `otx/mpa/modules/datasets/cls_dir_dataset.py`? For my best knowledge, classification task is using `otx/algorithms/classification/adapters/mmcls/data/datasets.py`.

File: otx/mpa/modules/datasets/cls_dir_dataset.py [hunk 3]
```diff
@@ -186,6 +198,9 @@ def evaluate(self, results, metric="accuracy", metric_options=None, logger=None)
...
+            eval_results[f"accuracy_top-{k}"] /= 100
```
Comment (JihwanEom) [inline otx/mpa/modules/datasets/cls_dir_dataset.py]: How about 
`eval_results = {key:val/100 if key.startswith('accuracy_top-') else {key:val} for key, val in eval_results.items()}`?

**Review by JihwanEom — CHANGES_REQUESTED** [review]

File: otx/mpa/modules/datasets/cls_dir_dataset.py (hunk 3 above)
Comment (sovrasov) [reply otx/mpa/modules/datasets/cls_dir_dataset.py]: @JihwanEom we were discussing the future otx code guideline recently (one of the candidates is https://google.github.io/styleguide/pyguide.html#power-features), lets avoid hard-to-read dict comprehensions with conditions.

Comment (JihwanEom) [reply otx/mpa/modules/datasets/cls_dir_dataset.py]: Okay, thank you for kind explanation!

File: otx/mpa/modules/datasets/cls_dir_dataset.py (hunk 2 above)
Comment (sovrasov) [reply otx/mpa/modules/datasets/cls_dir_dataset.py]: @JihwanEom @kprokofi If we don't use this dataset, why do we need this file?

Comment (kprokofi) [reply otx/mpa/modules/datasets/cls_dir_dataset.py]: Agree, maybe we can consider deleting all the unnecessary files?

Comment (JihwanEom) [reply otx/mpa/modules/datasets/cls_dir_dataset.py]: It will be removed soon; needed for MPA inner source repository. 
How about remove unnecessary files in other PR? We can clear unused files while writing unit tests.

Comment (kprokofi) [reply otx/mpa/modules/datasets/cls_dir_dataset.py]: Yeah, in other PRs, of cource

File: otx/algorithms/classification/adapters/mmcls/data/datasets.py [hunk 4]
```diff
@@ -149,6 +149,9 @@ def evaluate(
             self.class_acc = True
 
         eval_results = super().evaluate(results, metrics, metric_options, logger=logger)
+        for k in metric_options["topk"]:
+            eval_results[f"accuracy_top-{k}"] /= 100
+            assert 0 <= eval_results[f"accuracy_top-{k}"] <= 1
```
Comment (JihwanEom) [inline otx/algorithms/classification/adapters/mmcls/data/datasets.py]: How about using `logger.warning` instead of `assert`? If `accuracy-top-1` is given as `nan`, otx will die.

File: otx/mpa/modules/datasets/cls_dir_dataset.py [hunk 5]
```diff
@@ -33,7 +33,15 @@ class ClsDirDataset(BaseDataset):
         use_labels (bool): dataset with labels or unlabels
     """
 
-    def __init__(self, data_dir, pipeline=[], classes=[], new_classes=[], use_labels=True, **kwargs):
+    def __init__(
```
Comment (JihwanEom) [inline otx/mpa/modules/datasets/cls_dir_dataset.py]: Could you revert changes on this file? Even though it will be removed soon, but it can cause some confusion for other members.

File: otx/algorithms/classification/adapters/mmcls/data/datasets.py (hunk 4 above)
Comment (kprokofi) [reply otx/algorithms/classification/adapters/mmcls/data/datasets.py]: is `nan` metric expected? I think it is not normal case if we have nan in metric

File: otx/mpa/modules/datasets/cls_dir_dataset.py (hunk 5 above)
Comment (kprokofi) [reply otx/mpa/modules/datasets/cls_dir_dataset.py]: Ok. I just didn't want to spend time reverting this file as we don't use it anymore. But, I agree that it can lead to unnecessary changes in commits.

File: otx/algorithms/classification/adapters/mmcls/data/datasets.py (hunk 4 above)
Comment (JihwanEom) [reply otx/algorithms/classification/adapters/mmcls/data/datasets.py]: Yes, as you said, `nan` is not expected for normal situation. But I believe even though `nan` is occurred in training, otx should be alive. 
How about `raise`? In my memory, we agreed to avoid `assert` in our product. @harimkang @sungmanc Could you share your thoughts?

Comment (kprokofi) [reply otx/algorithms/classification/adapters/mmcls/data/datasets.py]: I think we don't need this check at all. It is useless. If the accuracy come from mmcls in the range from [0,1] we wouldn't catch it anyway. I deleted it

**Comment by JihwanEom** [comment]
Could you check test results?

**Comment by kprokofi** [comment]
> Could you check test results?

Code-quality passed, there was problem not related to my PR
Pre-Merge tests are failed for every task, again, not related to this PR

**Comment by JihwanEom** [comment]
> > Could you check test results?
> 
> Code-quality passed, there was problem not related to my PR Pre-Merge tests are failed for every task, again, not related to this PR

Nice! Could you check also tests/integration/cli and tests/unit locally?

**Comment by sungmanc** [comment]
> Nice! Could you check also tests/integration/cli and tests/unit locally?

Before conducting the local test, please remind that pull the latest OTX commit. It clearly reduces the test time.

**Comment by JihwanEom** [comment]
CI is working now, could you rebase and run tests?

**Comment by kprokofi** [comment]
As I can see errors are not related to my PR:
image
The first one related to timeout issue during segmentation training
image
The second one unexpected key in ResNet backbone initialization:
image
The last one related to TestInstanceSegmentationCLI

We need to fix it in another PRs

**Comment by kprokofi** [comment]
@JihwanEom Could you approve if no objection other tests are OK

**Review by JihwanEom — APPROVED** [review]
Thank you for hard work! LGTM.

**Review by sovrasov — APPROVED** [review]
